src/compiler/parser.ts

```diff
@@ -436,6 +436,13 @@ module ts {
         // attached to the EOF token.
         let parseErrorBeforeNextFinishedNode: boolean = false;
 
+        export const enum StatementFlags {

```

Comment:
No, your first interpretation of my question was the correct one.

If these flags are about the statements themselves (as opposed to the context), then why does parseModuleElementOfKind take a StatementFlags instead of a ParsingContext? I thought that parameter represents what we are trying to parse, not a property of the statement.